src/sentry/data_export/endpoints/data_export.py

```diff
@@ -47,7 +47,11 @@ def post(self, request, organization):
             return Response(serializer.errors, status=400)
 
         data = serializer.validated_data
-
+        # Ensure discover features are enabled if necessary
+        if data["query_type"] == ExportQueryType.DISCOVER_STR and not features.has(
+            "organizations:discover-basic", organization, actor=request.user
+        ):
+            return Response({"detail": "You do not have access to discover features"}, status=403)
```

Comment:
I think we should add a test for this as well, we can for now just test getting back that 403

Reply:
Noted! I'll add the test in https://github.com/getsentry/sentry/pull/17603